### PR TITLE
ML: Add segmenting on images or video and add a way to playback with learned classifiers.

### PR DESCRIPTION
**Pull request by mattlangford**
I plan to allow either sklearn classifiers or opencv classifiers shortly.

I didn't get to test these with bag files, so I hope the changes I made didn't æffect the already working stuff.

**TODO**: Make an interface to work with different learning algorithms in order to make them easier to interchange. This will allow us to test different methods and parameters easier. It will also keep track of kernels (#155).

### CODE REVIEW

File: gnc/sub8_perception/sub8_vision_tools/labelling/manual_grabcut.py [hunk 1]
```diff
@@ -188,32 +189,46 @@ def save(self, mask):
...
         print 'Creating new pickle'
         data = []
     else:
         data = pickle.load(open(args.append, "rb"))
 
-    print bc.image_topics[0]
-    print bc.image_topics[0]
+    file_name = args.file_name
+    if file_name.split('.')[-1] == 'bag':
+        import rospy
```
Comment (jpanikulam) [inline gnc/sub8_perception/sub8_vision_tools/labelling/manual_grabcut.py:216]: Is it common to not have ros installed?

File: gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import argparse
+import cv2
+import pickle
+import numpy as np
+import sys
+from adaboost import observe
+
+if __name__ == '__main__':
+    usage_msg = ("Pass the path to a bag, the start of an image sequence,or 'video' for a webcam to play through the media and look for the learned object.")
+    desc_msg = "glhf"
```
Comment (jpanikulam) [inline gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py:10]: Nbd: Desc messages are optional if you don't want one

File: gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    last_mask = None
+    if args.topic is not None:
+        assert args.topic in bc.image_topics, "{} not in the bag".format(args.topic)
+        print 'Crawling topic {}'.format(args.topic)
+        crawl = bc.crawl(topic=args.topic)
+    else:
+        crawl = bc.crawl(topic=bc.image_topics[0])
+
+    for image in crawl:
```
Comment (jpanikulam) [inline gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py:42]: Glad you made this script

File: gnc/sub8_perception/sub8_vision_tools/labelling/manual_grabcut.py (hunk 1 above)
Comment (mattlangford) [inline gnc/sub8_perception/sub8_vision_tools/labelling/manual_grabcut.py:216]: I just got a new mac lol
I'm scared to try

Comment (jpanikulam) [inline gnc/sub8_perception/sub8_vision_tools/labelling/manual_grabcut.py:216]: Hahahahahahaha

**Comment by mattlangford** [comment]
I will probably add to this tonight again if you haven't merged by then

**Comment by jpanikulam** [comment]
I approve this PR. If you want to add stuff before merging, that works too

File: gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py (hunk 2 above)
Comment (mattlangford) [inline gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py:10]: I tried to remove it and if I -h, i get an error

Comment (jpanikulam) [inline gnc/sub8_perception/sub8_vision_tools/machine_learning/playback.py:10]: Did you also remove it in the call to the Parser constructor?

**Comment by jpanikulam** [comment]
Needs rebase from upstream, otherwise good

**Pull request merged**